zap/src/main/java/org/zaproxy/zap/spider/Spider.java

```diff
@@ -209,9 +209,9 @@ private void init() {
             this.addFetchFilter(filter);
         }
 
-        // Add a default parse filter and any custom ones
-        this.addParseFilter(new DefaultParseFilter(spiderParam, extension.getMessages()));
+        // Add custom parse filters and set a default one
         for (ParseFilter filter : extension.getCustomParseFilters()) this.addParseFilter(filter);
+        this.setDefaultParseFilter(new DefaultParseFilter(spiderParam, extension.getMessages()));
```

Comment:
- This could call the controller directly, we don't need to expose in the public API for now.
 - Could be done in the previous line as it no longer needs to be done last.